### PR TITLE
Use createRoot in ReactTestUtilsActUnmockedScheduler-test

### PR DESCRIPTION
**Pull request by jackpope**

### CODE REVIEW

File: packages/react-dom/src/__tests__/ReactTestUtilsActUnmockedScheduler-test.js [hunk 1]
```diff
@@ -23,48 +24,43 @@ function clearLog() {
...
-  ReactDOM.render(el, dom);
-}
-
-function unmount(dom) {
-  ReactDOM.unmountComponentAtNode(dom);
-}
-
 beforeEach(() => {
+  prevActGlobal = global.IS_REACT_ACT_ENVIRONMENT;
+  global.IS_REACT_ACT_ENVIRONMENT = true;
```
Comment (rickhanlonii) [inline packages/react-dom/src/__tests__/ReactTestUtilsActUnmockedScheduler-test.js:29]: This test is testing the unmocked scheduler (from the file name) so you can't use the internal act here.

**Review by rickhanlonii — CHANGES_REQUESTED** [review]

File: packages/react-dom/src/__tests__/ReactTestUtilsActUnmockedScheduler-test.js (hunk 1 above)
Comment (rickhanlonii) [reply packages/react-dom/src/__tests__/ReactTestUtilsActUnmockedScheduler-test.js:29]: You should be able to use the public act here though.

Comment (rickhanlonii) [inline packages/react-dom/src/__tests__/ReactTestUtilsActUnmockedScheduler-test.js:29]: Wait why is this needed now and not before?

Comment (jackpope) [reply packages/react-dom/src/__tests__/ReactTestUtilsActUnmockedScheduler-test.js:29]: With concurrent rendering there is a new path hit during the setState https://github.com/facebook/react/blob/main/packages/react-reconciler/src/ReactFiberWorkLoop.js#L4006

isConcurrentActEnvironment logs and error if the global is not set

Comment (rickhanlonii) [reply packages/react-dom/src/__tests__/ReactTestUtilsActUnmockedScheduler-test.js:29]: Oh, right, this is required to use external act.

**Review by rickhanlonii — APPROVED** [review]
